File: CS/CustomWizardExample/Wizard/CustomDataSourceCreationService.cs
using DevExpress.Snap.Services;
using DevExpress.DataAccess.Wizard.Model;

namespace CustomWizardExample.Wizard {
    public class CustomDataSourceCreationService : DataSourceCreationService {
        public override DevExpress.DataAccess.IDataComponent CreateDataComponent(IDataSourceModel model) {
            SnapDataSourceModel dataSourceModel = (SnapDataSourceModel)model;
            if (dataSourceModel.DataSourceType == CustomDataSourceType.XmlPersons)
                return new XmlPersonDataComponent((string)dataSourceModel.Tag) { Name = dataSourceModel.DataSourceName };
            return base.CreateDataComponent(model);
        }
    }
}

File: VB/CustomWizardExample/Wizard/CustomDataSourceCreationService.vb
Imports DevExpress.Snap.Services
Imports DevExpress.DataAccess.Wizard.Model

Namespace CustomWizardExample.Wizard
    Public Class CustomDataSourceCreationService
        Inherits DataSourceCreationService

        Public Overrides Function CreateDataComponent(ByVal model As IDataSourceModel) As DevExpress.DataAccess.IDataComponent
            Dim dataSourceModel As SnapDataSourceModel = DirectCast(model, SnapDataSourceModel)
            If dataSourceModel.DataSourceType Is CustomDataSourceType.XmlPersons Then
                Return New XmlPersonDataComponent(CStr(dataSourceModel.Tag)) With {.Name = dataSourceModel.DataSourceName}
            End If
            Return MyBase.CreateDataComponent(model)
        End Function
    End Class
End Namespace
